src/test/java/eu/europeana/metis/sandbox/service/workflow/HarvestServiceImplTest.java

```diff
@@ -0,0 +1,82 @@
+package eu.europeana.metis.sandbox.service.workflow;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import eu.europeana.metis.sandbox.common.exception.ServiceException;
+import eu.europeana.metis.utils.ZipFileReader;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.mock.web.MockMultipartFile;
+
+
+@ExtendWith(MockitoExtension.class)
+public class HarvestServiceImplTest {
+
+  private static final ZipFileReader zipfilereader = new ZipFileReader();
+
+  private static final HarvestServiceImpl harvestService = new HarvestServiceImpl();
+
+  @Test
+  void harvestServiceFromURL_expectSuccess() throws IOException {
+
+    Path dataSetPath = Paths.get("src", "test", "resources", "zip", "dataset-valid.zip");
+    assertTrue(Files.exists(dataSetPath));
+
+    var expectedRecords = zipfilereader.getContentFromZipFile(Files.newInputStream(dataSetPath));
+
+    var records = harvestService.harvest(dataSetPath.toUri().toString());
+
+    // In spite records being the same size, they are not in the same order.
```

Comment:
I'm assuming you wanted to test the size of list of bytes for each record?
If you do want that, one way to do this could be perhaps creating a set of all records sizes for both the expected and retrieved, and the items of those should be identical.
(order is not important in this case)

Reply:
Correct. It's now implemented